src/lib/components/quiz/quiz.svelte

```diff
@@ -0,0 +1,249 @@
+<script lang="ts">
+  import type { QuizData, TeamMatch } from '$lib/constants/quiz';
+  import LeftArrow from '$lib/components/icons/left-arrow.svelte';
+  import RightArrow from '$lib/components/icons/right-arrow.svelte';
+  import MoreInfo from './more-info.svelte';
+
+  export let data: QuizData;
+
+  $: index = 0;
+  let response: TeamMatch[] = [];
+  $: answeredAllQuestions =
+    response.length === data.questions.length && !response.includes(undefined);
+  let showResults = false;
+  let showMoreInfo = false;
+  let showTeam: string;
+  $: talliedResponses = response.reduce((tallies, match) => {
+    if (tallies[match]) tallies[match]++;
+    else tallies[match] = 1;
+    return tallies;
+  }, {});
+  let teams = ['Algo', 'Dev', 'Design', 'AI'];
+  let match: string;
+
+  function goLeft() {
+    if (index > 0) {
+      index--;
+    }
+  }
+
+  function goRight() {
+    if (index < data.questions.length - 1) {
+      index++;
+    }
+    if (index === data.questions.length) {
+      // need to display text at the bottom that says:
+      console.log('please answer all questions');
+    }
+  }
+
+  function recordAnswer(match: TeamMatch) {
+    console.log(match);
+
+    response[index] = match;
+    // if answered all questions and no more questions then submitResponses
+    if (!answeredAllQuestions) {
+      goRight();
+    }
+  }
+
+  function submitResponses() {
+    let totalTallyKey: string[] = Object.keys(talliedResponses);
+    let totalTallyValue: number[] = Object.values(talliedResponses);
+    match = totalTallyKey[totalTallyValue.indexOf(Math.max(...totalTallyValue))];
+    console.log(match);
+    // save to localstorage
+    // show results and hide the question
+    showResults = true;
+  }
+
+  function restartQuiz() {
+    response = [];
+    index = 0;
+    showResults = false;
+  }
+
+  function showTeamDetails(currentTeam: string) {
+    showMoreInfo = true;
+    showTeam = currentTeam;
+  }
+  function goBackToResults() {
+    showMoreInfo = false;
+  }
+</script>
+
+<div class="container">
+  <!-- DISPLAY THE QUIZ QUESTIONS -->
+  {#if !showResults}
+    <h1>ACM PATH QUIZ</h1>
+    <div class="question">
+      <h2>{data.questions[index].prompt}</h2>
+      {#if response[index]}
+        <p>answered</p>
+      {/if}
```

Comment:
They can update their responses already, this is merely a quick fix to let them know that they already answered the question just in case they forgot. I can see why this is misleading so don't resolve this issue I will come up with a better UI solution.

Reply:
Done, now there are unique styles for selected answers